pybossa.js

```diff
@@ -189,14 +196,12 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
             return data;});
     }
 
-    function _getCurrentTaskId(url) {
-        pathArray = url.split('/');
-        if (url.indexOf('/task/')!=-1) {
-            var l = pathArray.length;
-            var i = 0;
-            for (i=0;i<l;i++) {
-                if (pathArray[i]=='task') {
-                    return pathArray[i+1];
+    function _getCurrentTaskId(pathname) {
```

Comment:
Is the refactoring of this method for a specific change or styling?

Reply:
The reason I ended up refactoring this function was because initially, I intended to modify this function to satisfy our need. But it ended up not necessary. So since I have fully reviewed this function, it worths doing some refactoring otherwise we will never have the change to refactor it.

The refactoring here is mainly styling. But changing the parameter name from url to pathname potentially reduces future bugs. Because passing a URL with query parameter to this function might end up undesired behavior. And the way this function was used, it actually accepts the "pathname".

I understand refactoring is a gradual work. Also, sometimes it's dangerous of course. I usually do some refactoring on the code I'm touching, which I think it should reduce the risk while keep quality improving.